lib/output_helper.rb

```diff
@@ -92,15 +92,16 @@ def self.csv_line(vrp, service, cluster_index, two_stages = false)
 
   class Result
     def self.generate_geometry(solution)
-      return nil unless solution.to_h[:result]
+      return nil unless solution&.key?(:result)
 
       @colors = ['#DD0000', '#FFBB00', '#CC1882', '#00CC00', '#558800', '#009EFF', '#9000EE',
                  '#0077A3', '#000000', '#003880', '#BEE562']
 
       expected_geometry = solution[:configuration].to_h[:geometry].to_a.map!(&:to_sym) # TODO : investigate how it is possible that no configuration is returned
+      return nil unless expected_geometry.any?
 
       expected_geometry.map!(&:to_sym)
-      [solution[:result]].flatten(1).collect{ |result|
+      (solution[:result].is_a?(Array) ? solution[:result] : [solution[:result]]).collect{ |result|
```

Comment:
`solution[:result] = [solution[:result]] unless solution.nil? || solution[:result].nil? || solution[:result].is_a?(Array)`* before calling `present` inside the else condition so that this function can work without any problems and we don't need to repeat this operation in two places.

*: or something like this.